__tests__/Auth0Client.helpers.ts

```diff
@@ -56,16 +56,53 @@ export const setupFn = mockVerify => {
   };
 };
 
+const processDefaultLoginWithRedirectOptions = config => {
+  const defaultTokenResponseOptions = {
+    success: true,
+    response: {}
+  };
+  const defaultAuthorizeResponseOptions = {
+    code: TEST_CODE,
+    state: TEST_STATE
+  };
+  const token = {
+    ...defaultTokenResponseOptions,
+    ...(config.token || {})
+  };
+  const authorize = {
+    ...defaultAuthorizeResponseOptions,
+    ...(config.authorize || {})
+  };
+
+  return {
+    token,
+    authorize
+  };
+};
+
 export const loginWithRedirectFn = (mockWindow, mockFetch, fetchResponse) => {
   return async (
     auth0,
     options: RedirectLoginOptions = undefined,
-    tokenSuccess = true,
-    tokenResponse = {},
-    code = TEST_CODE,
-    state = TEST_STATE,
-    error = null
+    testConfig: {
+      token?: {
+        success?: boolean;
+        response?: any;
+      };
+      authorize?: {
+        code?: string;
+        state?: string;
+        error?: string;
+      };
+    } = {
+      token: {},
+      authorize: {}
+    }
```

Comment:
I added an object here instead of seperate parameters.
This makes the function itself a bit more complicated but it is a bit easier to call.